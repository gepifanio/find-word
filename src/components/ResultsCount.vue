<template>
  <p data-test="results-count">{{ count }} results</p>
</template>

<script>
export default {
  props: {
    count: {
      type: Number,
      default: 0
    }
  }
};
</script>
