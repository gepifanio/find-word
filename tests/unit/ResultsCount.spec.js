import { shallowMount } from '@vue/test-utils';
import ResultsCount from '@/components/ResultsCount.vue';

describe('ResultsCount.vue', () => {
  it('should match snapshot', () => {
    const result = shallowMount(ResultsCount, {
      propsData: {
        count: 2
      }
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('renders props.count when passed', () => {
    const count = 2;
    const wrapper = shallowMount(ResultsCount, {
      propsData: {
        count
      }
    });

    expect(typeof wrapper.props().count).toBe('number');
    expect(wrapper.props().count).toBe(count);
    expect(wrapper.find('[data-test="results-count"]').text()).toEqual(
      '2 results'
    );
  });
});
