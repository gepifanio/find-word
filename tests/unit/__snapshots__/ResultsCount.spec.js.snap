// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsCount.vue should match snapshot 1`] = `
<p
  data-test="results-count"
>
  2 results
</p>
`;
